src/_zkapauthorizer/tests/test_pricecalculator.py

```diff
@@ -222,3 +223,65 @@ def test_one_pass(self, pass_value: int) -> None:
             price,
             Equals(1),
         )
+
+    @given(
+        integers(min_value=2, max_value=255),
+        integers(min_value=0, max_value=254),
+    )
+    def test_minimum_spending(self, needed: int, extra_shares: int) -> None:
+        """
+        The minimum amount of spending must be at least the number
+        of 'required' shares
+        """
+        # ZFEC only allows up to 256 total shares
+        assume(needed + extra_shares < 256)
+
+        # "total" shares is encoded this way to give hypothesis a
+        # break: we know "total" must be >= "needed" so we just add
+        # some extra shares (possibly 0).
+        calculator = PriceCalculator(
+            pass_value=1000,
+            shares_needed=needed,
+            shares_total=needed + extra_shares,
+        )
+        price = calculator.calculate([1000])
+        self.assertThat(price, greater_or_equal(needed))
+
+    @given(
+        integers(min_value=1, max_value=100).flatmap(
+            lambda n_shares: lists(sizes(), min_size=n_shares, max_size=n_shares)
+        ),
```

Comment:
```suggestion
        lists(sizes(), min_size=1, max_size=100),
```

If `lists` only supported a `size=...` argument then the flatmap construction here would be useful but since it supports the same kind of size bounds as `integers` you don't really need it.